src/liballoc/arc.rs

```diff
@@ -200,16 +238,13 @@ impl<T: ?Sized> Arc<T> {
     /// # Examples
     ///
     /// ```
-    /// #![feature(arc_weak)]
-    ///
     /// use std::sync::Arc;
     ///
     /// let five = Arc::new(5);
     ///
     /// let weak_five = five.downgrade();
     /// ```
-    #[unstable(feature = "arc_weak",
-               reason = "Weak pointers may not belong in this module.")]
+    #[stable(feature = "arc_weak", since = "1.4.0")]
     pub fn downgrade(&self) -> Weak<T> {
```

Comment:
I would very much prefer this to take `this: &Arc<T>` like the other functions - implementing methods on a smart pointer causes really unfortunate behavior if `T` has a method called `downgrade`. `clone` is a special case since the entire point of `Arc` and `Rc` are to replace the behavior of `clone`.